### PR TITLE
Widgetfactory classify polygon

### PR DESCRIPTION
**Pull request by hennie-k**

### CODE REVIEW

File: geest/gui/widgets/classify_poly_into_classes_widget.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+    def update_fields(self, layer):
+        print(
+            f"[ClassifyPolyIntoClassesWidget] Updating fields for layer: {layer.name() if layer else 'None'}"
+        )
+        self.field_selector.clear()
+        if isinstance(layer, QgsVectorLayer):
+            fields = [field.name() for field in layer.fields()]
+            self.field_selector.addItems(fields)
+            print(f"[ClassifyPolyIntoClassesWidget] Fields added: {fields}")
```
Comment (timlinux) [inline geest/gui/widgets/classify_poly_into_classes_widget.py:53]: Do not use print please. Use The QgsLogMessage class

File: geest/gui/widgets/classify_poly_into_classes_widget.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
+from qgis.PyQt.QtWidgets import QWidget, QVBoxLayout, QLabel, QComboBox
+from qgis.PyQt.QtCore import pyqtSignal
+from qgis.gui import QgsMapLayerComboBox
```
Comment (timlinux) [inline geest/gui/widgets/classify_poly_into_classes_widget.py:3]: QgsFieldComboBox

File: geest/gui/widgets/classify_poly_into_classes_widget.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        self.setProperty("widget_type", "classify_poly_into_classes")
+
+        # Layer selector
+        self.layer_label = QLabel("Select Polygon Layer:")
+        self._layout.addWidget(self.layer_label)
+        self.layer_selector = QgsMapLayerComboBox()
+        self.layer_selector.setFilters(QgsMapLayerProxyModel.PolygonLayer)
+        self._layout.addWidget(self.layer_selector)
+
+        # Field selector
```
Comment (timlinux) [inline geest/gui/widgets/classify_poly_into_classes_widget.py:26]: QgsFieldComboBox

File: geest/gui/widgets/classify_poly_into_classes_widget.py (hunk 1 above)
Comment (hennie-k) [reply geest/gui/widgets/classify_poly_into_classes_widget.py]: Changes made and tested

File: geest/gui/widgets/classify_poly_into_classes_widget.py (hunk 2 above)
Comment (hennie-k) [reply geest/gui/widgets/classify_poly_into_classes_widget.py]: Changes made and tested

File: geest/gui/widgets/classify_poly_into_classes_widget.py (hunk 3 above)
Comment (hennie-k) [reply geest/gui/widgets/classify_poly_into_classes_widget.py:26]: Changes made and tested

**Pull request merged**